### PR TITLE
Add yast2_apparmor.pm

### PR DESCRIPTION
**Pull request by Zaoliang**
http://e13.suse.de/tests/1055#live

### CODE REVIEW

File: tests/console/yast2_apparmor.pm [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    # check Apparmor Configuration is opened
+    assert_screen 'yast2_apparmor';
+    send_key 'ret';
+    assert_screen 'yast2_apparmor_enabled';
+
+    # part 1: open profile mode configuration and check toggle/show all profiles
+    send_key 'alt-n';
+    assert_screen 'yast2_apparmor_profile_mode_configuration';
+    send_key 'alt-o';
+    wait_still_screen;
```
Comment (okurz) [inline tests/console/yast2_apparmor.pm]: `wait_still_screen` by default waits for 7 seconds. Does it also work with less? E.g. 1 or 2 seconds? (that would be `wait_still_screen(1);`

File: tests/console/yast2_apparmor.pm [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    send_key 'spc';
+    send_key 'alt-o';
+    wait_still_screen;
+    send_key 'alt-d';
+
+    # confirm to save changes to the profile
+    for (assert_screen 'yast2_apparmor_configuration_manage_existing_profiles_edit_file_changed') { send_key 'alt-y'; }
+
+    # close now AppArmor configuration
+    sleep 3;
```
Comment (okurz) [inline tests/console/yast2_apparmor.pm]: can you replace the sleeps with something more adaptive?

File: tests/console/yast2_apparmor.pm [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    send_key 'up';
+    send_key 'spc';
+    send_key 'down';
+    send_key 'spc';
+    send_key 'alt-o';
+    wait_still_screen;
+    send_key 'alt-d';
+
+    # confirm to save changes to the profile
+    for (assert_screen 'yast2_apparmor_configuration_manage_existing_profiles_edit_file_changed') { send_key 'alt-y'; }
```
Comment (okurz) [inline tests/console/yast2_apparmor.pm]: what should this "for(assert_screen)" mean? assert_screen does never return false

File: tests/console/yast2_apparmor.pm [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    # close now AppArmor configuration
+    sleep 3;
+    send_key 'alt-n';
+    sleep 2;
```
Comment (okurz) [inline tests/console/yast2_apparmor.pm]: and this

File: tests/console/yast2_apparmor.pm [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    wait_still_screen;
+
+    # Add entry for new profile
+    send_key 'alt-a';
+    send_key 'alt-r';
+    send_key 'alt-s';
+    send_key 'alt-d';
+    # confirm to save changes to profile
+    for (assert_screen 'yast2_apparmor_configuration_manage_existing_profiles_edit_file_changed') { send_key 'alt-y'; }
+    sleep 2;
```
Comment (okurz) [inline tests/console/yast2_apparmor.pm]: and this

File: tests/console/yast2_apparmor.pm [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    send_key 'alt-o';
+    wait_still_screen;
+
+    # Add entry for new profile
+    send_key 'alt-a';
+    send_key 'alt-r';
+    send_key 'alt-s';
+    send_key 'alt-d';
+    # confirm to save changes to profile
+    for (assert_screen 'yast2_apparmor_configuration_manage_existing_profiles_edit_file_changed') { send_key 'alt-y'; }
```
Comment (okurz) [inline tests/console/yast2_apparmor.pm]: see above

Comment (Zaoliang) [inline tests/console/yast2_apparmor.pm]: @okurz okay, will change sleep into wait_still_screen and define seconds for it as well.

File: tests/console/yast2_apparmor.pm (hunk 3 above)
Comment (coolo) [inline tests/console/yast2_apparmor.pm]: this is just invalid syntax. for will just iterate over the object returned from assert_screen once. I.e. the for is useless.

```
coolo@nohostname#~>perl -e 'for (1) { print "Hallo\n"; }'
Hallo
```

**Comment by coolo** [comment]
this needs a rebase and the rebase isn't easy (I tried myself :) as the yast tests moved to a different function in master :)

**Comment by okurz** [comment]
does it work now consistently many times in a row?

**Comment by Zaoliang** [comment]
@okurz yes. the test works well, finally :)

**Pull request merged**